packages/frontend/app/(dashboard)/course/[cid]/queue/[qid]/components/QuestionCard.tsx

```diff
@@ -59,141 +63,192 @@ const QuestionCard: React.FC<QuestionCardProps> = ({
-        </Col>
-        {isBeingHelped && !isStaff && question.helpedAt && (
-          <Col flex="0 0 3rem">
-            <div className="text-sm font-medium text-green-700">
-              {servedTime}
-            </div>
+            {question.questionTypes?.map((questionType, index) => (
+              <QuestionTagElement
+                key={index}
+                tagName={questionType.name}
+                tagColor={
+                  question.status !== LimboQuestionStatus.ReQueueing
+                    ? questionType.color
+                    : '#f0f0f0'
+                }
+              />
+            ))}
           </Col>
-        )}
-        <Col flex="0 0 3rem">
-          <div className="text-sm text-gray-600">{getWaitTime(question)}</div>
-          {isStaff && isBeingHelped && (
-            <div className="text-sm font-medium text-green-700">
-              {servedTime}
-            </div>
+          {isBeingHelped && !isStaff && question.helpedAt && (
+            <Col flex="0 0 3rem">
+              <div className="text-sm font-medium text-green-700">
+                {servedTime}
+              </div>
+            </Col>
           )}
-        </Col>
-        {isStaff && (
-          <Col className="w-full sm:w-auto">
-            <TAQuestionCardButtons
-              courseId={cid}
-              queueId={qid}
-              question={question}
-              hasUnresolvedRephraseAlert={false}
-              tasksSelectedForMarking={tasksSelectedForMarking}
-              className="align-center flex items-center justify-around"
-            />
+          <Col flex="0 0 3rem">
+            <div className="text-sm text-gray-600">{getWaitTime(question)}</div>
+            {isStaff && isBeingHelped && (
+              <div className="text-sm font-medium text-green-700">
+                {servedTime}
+              </div>
+            )}
           </Col>
-        )}
-        <div
-          className={`absolute left-auto right-1 ${question.text && question.questionTypes && question.questionTypes.length > 0 ? '-mt-[4.4rem]' : '-mt-12 md:-mt-[3.2rem]'}`}
-        >
-          {isBeingHelped && !isStaff && (
-            <div className="text-sm text-green-700">Currently Being Served</div>
+          {question.status !== LimboQuestionStatus.ReQueueing ? (
+            isStaff ? (
+              <Col className="w-full sm:w-auto">
+                <TAQuestionCardButtons
+                  courseId={cid}
+                  queueId={qid}
+                  question={question}
+                  hasUnresolvedRephraseAlert={false}
+                  tasksSelectedForMarking={tasksSelectedForMarking}
+                  className="align-center flex items-center justify-around"
+                />
+              </Col>
+            ) : null
+          ) : (
+            <Col className="w-full sm:w-auto">
+              <div className="text-md ml-0 h-full italic text-gray-600 sm:ml-2">
+                In the process of requeuing...
```

Comment:
Doing this kinda breaks the column of times (looks especially weird for students).
![image](https://github.com/user-attachments/assets/4d816cba-0c21-4504-92c0-580d0de5d01c)
![image](https://github.com/user-attachments/assets/97e1aa60-7476-4de2-b28d-fbf7753ceb4e)

On mobile, this text ends up ends up getting wrapped around, causing the question card to grow larger, which is not really ideal since i'd argue we're wanting to draw less attention to these types of questions and that increasing their size gives them more attention.
![image](https://github.com/user-attachments/assets/820a258a-4174-4ec0-a6e5-1ece04f4186e)
Though surprisingly it doesn't look too bad on the TA-side of things
![image](https://github.com/user-attachments/assets/22ca1202-aba5-4818-b329-5429a9371d48)

Nonetheless, I'd recommend doing the floating text instead of this like I had mentioned in my original comment, as that will kinda fix these issues with lining things up and consistency. And by floating text i mean like the "currently being helped text" but instead it says "Not Ready" or "Requeueing..." or something. As much as the floating text isn't ideal, it's kinda the only solution as conditionally adding anything to a question card is surprisingly hard to do while still making everything look consistent and well-aligned.